### PR TITLE
Fix/vote count

### PR DESCRIPTION
**Pull request by mithun30052001**
In a particular post/comment, when the user clicks the upvote and then clicks the downvote button, the number of votes displayed should be the original number of votes-1 #108  issue resolved

### CODE REVIEW

File: app/controllers/votes_controller.rb [hunk 1]
```diff
@@ -12,15 +12,25 @@ def create
...
           end
         end
+      else
+        if vote.save
+          @success = true
+        else
+          @success = false
+        end
+      end
+
```
Comment (iMagesh) [inline app/controllers/votes_controller.rb:33]: can you refactor this code? there seems to be too many if else statements

**Comment by iMagesh** [comment]
@mithun30052001 resolve conflicts

**Comment by mithun30052001** [comment]
the conflicts have been resolved

File: app/controllers/votes_controller.rb [hunk 2]
```diff
@@ -12,26 +12,8 @@ def create
...
-        if vote.save
-          @success = true
-        else
-          @success = false
-        end
-      end
-
-
+        existing_vote.first.destroy if existing_vote.size > 0
+        vote.save unless existing_vote.size > 0 && existing_vote.first.upvote == vote.upvote
```
Comment (iMagesh) [inline app/controllers/votes_controller.rb:16]: why do you have these logics under format.js ?

Comment (mithun30052001) [reply app/controllers/votes_controller.rb:16]: i tried to do these logics in model magesh but its not working magesh.It is working when we have it here only magesh

Comment (iMagesh) [reply app/controllers/votes_controller.rb:16]: > i tried to do these logics in model magesh but its not working magesh.It is working when we have it here only magesh

why not? it should work. This one right here is not a good practice, so figure it out.

Comment (mithun30052001) [reply app/controllers/votes_controller.rb:16]: This issue is being worked on by aboorva

**Comment by iMagesh** [comment]
@Aboorva1 is this done? can we merge? @mithun30052001

**Pull request closed without merging**